src/systems/systems/src/pneumatic/mod.rs

```diff
@@ -128,32 +218,53 @@ impl PneumaticPipe {
     }
 
     fn change_volume(&mut self, new_volume: Volume) {
+        let new_pressure = self.calculate_pressure_for_new_volume(new_volume);
+        self.update_temperature_for_target_pressure(new_pressure);
+        self.pressure = new_pressure;
         self.volume = new_volume;
     }
+
+    fn calculate_pressure_for_new_volume(&mut self, new_volume: Volume) -> Pressure {
+        self.pressure
+            * (self.volume.get::<cubic_meter>() / new_volume.get::<cubic_meter>())
+                .powf(Self::HEAT_CAPACITY_RATIO)
+    }
 }
 
-pub struct TargetPressureSignal {
+pub struct TargetPressureTemperatureSignal {
     target_pressure: Pressure,
+    target_temperature: ThermodynamicTemperature,
 }
```

Comment:
I like the idea of having a temperature here as well 👍.

Reply:
It's used because the air already has this temperature when it comes out of the engine.